### ISSUE TITLE
Switching Camera is not working 

### ISSUE THREAD

**Issue by Govind13100** [twilio/twilio-video.js#574]
Firstly i was using twilio-video.js version 1.14.0 in which the switch code : 

`Twilio.Video.createLocalVideoTrack({
      deviceId: { exact: camArray[camIndex] }
    }).then(function(LocalVideoTrackPublication) {
      console.log(LocalVideoTrackPublication);
      const tracks = Array.from(localParticipant.videoTracks.values());
      console.log(tracks);
      console.log(localParticipant);
      localParticipant.unpublishTracks(tracks);
      //console.log(localParticipant.identity   +" removed track: " +  tracks[0].kind);
      detachTracks(tracks);

      localParticipant.publishTrack(LocalVideoTrackPublication);
      //console.log(localParticipant.identity +  " added track: " +  localVideoTrack.kind);
      const previewContainer = document.getElementById('local-media');
      attachTracks([localVideoTrack], previewContainer);

    });`

works great! But when i used Twilio-video.js version 2.0.0-beta5, it throws the below error : 

`Uncaught (in promise) TypeError: track must be a LocalAudioTrack, LocalVideoTrack, LocalDataTrack, or MediaStreamTrack`

I don't understand what cause this error and even i don't found any document for this. 

I will appreciate any help from any one here 

Thanks 
Govind

**Comment by manjeshbhargav** [comment]
Hi @Govind13100 ,

It seems like you are experiencing some of the breaking changes introduced in 2.x versions. Please refer to this [guide](https://www.twilio.com/docs/video/migrating-1x-2x) in order to migrate your application to 2.x version of the SDK. In your specific example, you can access your existing tracks like this:
```js
const tracks = Array.from(localParticipant.tracks.values()).map(publication => publication.track);
localParticipant.unpublishTracks(tracks);
detachTracks(tracks);
```
Since this is an issue with your implementation and not an SDK issue, I will close this issue for now.

Thanks,

Manjesh Malavalli
JSDK Team

**Comment by Govind13100** [comment]
@manjeshbhargav 

Thanks for replying with your answer, that really worked. But when i switch the camera, the audio is not coming, suddenly audio got stopped, i think i have to publish audio track too with video , right ?

i have tried it but i am not getting any positive result, would please help me out here ?

will be very appreciating 
thanks

**Comment by manjeshbhargav** [comment]
@Govind13100 ,

You can make sure to unpublish only the video tracks like this:
```js
const tracks = Array.from(localParticipant.videoTracks.values()).map(publication => publication.track);
localParticipant.unpublishTracks(tracks);
detachTracks(tracks);
```
This way, the audio tracks are still being published.

Thanks,

Manjesh Malavalli
JSDK Team

**Comment by niyasns** [comment]
@manjeshbhargav 
For switching camera, I tried the steps below
1. `createLocalVideoTrack` for creating new local video track
2. unpublish the old video track and published newly created video track (using the code in the last comment)
But after switching, the receiver can't hear the audio from my device. How to solve this?

At last, I used `createLocalTracks` to create a new video and audio tracks for switching. Then publish those new audio and video tracks to make the switch works without any issue on audio in the receiver. Do I need to create new audio and video tracks while switching the camera?
